app/parse.py

```diff
@@ -9,4 +14,26 @@ class Course:
 
 
 def get_all_courses() -> list[Course]:
-    pass
+    results = []
+    request_courses = requests.get(BASE_URL)
```

Comment:
Consider adding error handling for the HTTP request. If the request fails (e.g., due to network issues or a non-200 status code), it could cause the program to crash or behave unexpectedly.